src/views/CountryThreeWProjects/Map/index.tsx

```diff
@@ -0,0 +1,397 @@
+    getPointCircleHaloPaint,
+    pointColorMap,
+} from '#utils/map';
+import {
+    COLOR_RED,
+    COLOR_BLUE,
+    COLOR_ORANGE,
+    OPERATION_TYPE_EMERGENCY,
+    OPERATION_TYPE_MULTI,
+    OPERATION_TYPE_PROGRAMME,
+} from '#utils/constants';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type District = NonNullable<GoApiResponse<'/api/v2/district/'>['results']>[number];
+type Project = NonNullable<GoApiResponse<'/api/v2/project/'>['results']>[number];
+
+const redPointCirclePaint = getPointCirclePaint(COLOR_RED);
+const bluePointCirclePaint = getPointCirclePaint(COLOR_BLUE);
+const orangePointCirclePaint = getPointCirclePaint(COLOR_ORANGE);
+const sourceOption: mapboxgl.GeoJSONSourceRaw = {
+    type: 'geojson',
+};
+
+interface GeoJsonProps {
+    districtId: number;
+    numProjects: number;
+}
+
+interface ClickedPoint {
+    feature: GeoJSON.Feature<GeoJSON.Point, GeoJsonProps>;
+    lngLat: mapboxgl.LngLatLike;
+}
+
+type ProjectGeoJson = GeoJSON.FeatureCollection<GeoJSON.Point, GeoJsonProps>;
+
+function getOperationType(projectList: Project[]) {
+    const operationTypeList = unique(
+        projectList
+            .filter((d) => isDefined(d.operation_type))
+            .map((d) => ({
+                // FIXME: typings should be fixed in the server
+                id: d.operation_type as number,
+            })),
+        (d) => d.id,
+    ) ?? [];
+
+    if (operationTypeList.length === 0) {
+        return undefined;
+    }
+
+    if (operationTypeList.length === 1) {
+        return operationTypeList[0];
+    }
+
+    return {
+        id: OPERATION_TYPE_MULTI,
+    };
+}
+
+function getGeoJson(
+    districtList: District[],
+    districtDenormalizedProjectList: (Project & {
+        project_district_detail: Pick<
+            District,
+            'code'
+            | 'id'
+            | 'is_deprecated'
+            | 'is_enclave'
+            | 'name'
+        >;
+    })[],
+    requiredOperationTypeId: number,
+): ProjectGeoJson {
+    return {
+        type: 'FeatureCollection' as const,
+        features: districtList.map((district) => {
+            const projects = districtDenormalizedProjectList
+                .filter((project) => project.project_district_detail.id === district.id);
+
+            if (projects.length === 0) {
+                return undefined;
+            }
+
+            const operationType = getOperationType(projects);
+
+            if (operationType?.id !== requiredOperationTypeId) {
+                return undefined;
+            }
+
+            return {
+                id: district.id,
+                type: 'Feature' as const,
+                properties: {
+                    districtId: district.id,
+                    numProjects: projects.length,
+                },
+                geometry: {
+                    type: 'Point' as const,
+                    coordinates: district.centroid?.coordinates as [number, number] ?? [0, 0],
```

Comment:
Instead of adding a [0, 0] we should just remove the point.

cc: @frozenhelium

Reply:
using empty array for now

Reply:
We should not use empty array, the map will crash